urllib3/connectionpool.py

```diff
@@ -381,8 +382,22 @@ def _make_request(self, conn, method, url, timeout=_Default,
         log.debug("\"%s %s %s\" %s %s" % (method, url, http_version,
                                           httplib_response.status,
                                           httplib_response.length))
+
+        header_parsing_errors = extract_parsing_errors(httplib_response.msg)
```

Comment:
- Why do we prefer returning a value rather than raising an exception/catch-logging?
- Why return a namedtuple thing? This also feels like it should be an exception with a couple extra properties.